libvast/vast/detail/recursive_size.hpp

```diff
@@ -36,7 +36,9 @@ recursive_size(const std::filesystem::path& root_dir) {
     return caf::make_error(ec::filesystem_error, err.message());
   for (const auto& f : dir) {
     if (f.is_regular_file()) {
-      const auto size = f.file_size();
+      const auto size = f.file_size(err);
+      if (err)
+        return caf::make_error(ec::filesystem_error, err.message());
```

Comment:
This should not lead to an error.
```suggestion
        VAST_WARNING("vast is unable to get the size of {}: {}", f, err.message());
```

Reply:
That may be true in the case of the disk monitor, but is it true in the case of a function that determines the size of a directory?

Reply:
ok, lets deal with the toctou correctly:
```
if (err == std::make_error_code(std::errc::no_such_file_or_directory)) {
  VAST_WARN(..);
  continue;
} else {
  return make_error(...);
}
```

Reply:
Maybe this shouldn't even emit a warning?

Reply:
That's a nice solution. I agree on not warning. Will fix.

Reply:
Good! I'm not sure whether `make_error_code` is the correct way to check for a certain error, just fyi.

Reply:
Imho `VAST_WARN` is far too noisy for this, the "size" of a directory is an ephemeral, best-effort value anyways (the whole directory can be removed the millisecond after this function returns, files can be resized or deleted, etc.). If we want reliable values, we'd implement a different and much more complicated function using filesystem locking etc.

Reply:
Please check back @tobim